platform/ios/scripts/generate-style-code.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const fs = require('fs');
+const ejs = require('ejs');
+const spec = require('mapbox-gl-style-spec').latest;
+
+global.camelize = function (str) {
+    return str.replace(/(?:^|-)(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.camelizeWithLeadingLowercase = function (str) {
+    return str.replace(/-(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.isObject = function (property) {
+    switch (property.type) {
+        case 'string':
+            return true;
+        case 'color':
+            return true;
+        case 'array':
+            return true;
+        case 'enum': // (enum will be converted to NSString hence isObject(enum) == true)
+            return true;
+        default:
+            return false;
+    }
+}
+
+global.propertyType = function (property) {
+    switch (property.type) {
+        case 'boolean':
+            return 'BOOL';
+        case 'number':
+            return 'float';
+        case 'string':
+            return 'NSString';
+        case 'enum':
+            return 'NSString';
+        case 'color':
+            return `MGLColor`;
+        case 'array':
+            if (property.length) {
+                return `NSMutableArray`;
+            } else {
+                return `CGVector`;
```

Comment:
In that case, I think we should use `CGVector` for specific properties, based on the property name. The style specification uses its `Array` type for a variety of things that Cocoa conventionally represents with different types. For example:

* [`center`](https://www.mapbox.com/mapbox-gl-style-spec/#root-center) as `CLLocationCoordinate2D`
* [`fill-translate`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-fill-translate), [`line-translate`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-line-translate), [`icon-offset`](https://www.mapbox.com/mapbox-gl-style-spec/#layout-symbol-icon-offset), [`text-offset`](https://www.mapbox.com/mapbox-gl-style-spec/#layout-symbol-text-offset), [`icon-translate`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-icon-translate), [`text-translate`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-text-translate), [`circle-translate`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-circle-translate) as `CGVector`
  * `text-offset` is a good example of why `CGVector` is preferable to `NSPoint`/`CGPoint`: its unit is ems rather than screen points
* [`line-dasharray`](https://www.mapbox.com/mapbox-gl-style-spec/#paint-line-dasharray) as `NSArray<NSNumber *> *` or `CGFloat *`
  * I’m wary of introducing C arrays into this API. If we use NSNumber, maybe we should use NSNumber for [all the other `Number`-typed properties](https://github.com/mapbox/mapbox-gl-native/pull/5727/files/2ab2ab222fc4b4aa5eaaeaba7c57f0c535accb34#r71622482) too.
* [`icon-text-fit-padding`](https://www.mapbox.com/mapbox-gl-style-spec/#layout-symbol-icon-text-fit-padding) as `NSEdgeInsets`/`UIEdgeInsets`
* [`text-font`](https://www.mapbox.com/mapbox-gl-style-spec/#layout-symbol-icon-offset) as `NSArray<NSString *> *` (or `NSArray<NSFontDescriptor *> *` / `NSArray<UIFontDescriptor *> *`?)